### ISSUE TITLE
A nix derivation to build gfal2

### ISSUE THREAD

**Issue by v1kko** [cern-fts/gfal2#17]
I noticed  that a nix derivation was not yet available, so I created one (see below).

- It is quite minimal, but most things seem to work for me
- plugins for which no nix package is available are disabled
- I have issues with gfal2 + xrootd + proxy certificate, which gives the following error: 
  ```
  gfal-ls error: 52 (Invalid exchange) - [gfal2_stat][gfal_plugin_statG][gfal_xrootd_statG] Failed to stat file (Invalid exchange)
  ``` 

I hope this derivation can be of use to others as well, if anybody wants to try and merge it into [nixpkgs](https://github.com/NixOS/nixpkgs) Feel free to do so.

```nix
{ lib
, stdenv
, fetchFromGitHub
, cmake
, pkg-config
, glib
, dcap
, pcre2
, json_c
, doxygen
, pugixml
, davix-copy
, cryptopp
, libssh2
, xrootd
, gtest
, openldap
, libuuid
}:


stdenv.mkDerivation rec {
  pname = "gfal2";
  version = "2.22.0";

  src = fetchFromGitHub {
    owner = "cern-fts";
    repo = "${pname}";
    rev = "v${version}";
    sha256 = "198ql2qa9f8x77x6mhx0g004kyimvf45gy5ikwcx1v1p20avfs33";
  };

  nativeBuildInputs = [
    cmake
    pkg-config
    glib
    pcre2.dev
    json_c.dev
    doxygen
    pugixml
    cryptopp.dev
    gtest.dev
    openldap.dev
    libuuid.dev
    xrootd.dev
    davix-copy
    libssh2.dev
    dcap
  ];


  cmakeFlags = [
    "-DPLUGIN_LFC=FALSE"
    "-DPLUGIN_SRM=FALSE"
    "-DPLUGIN_RFIO=FALSE"
    "-DPLUGIN_GRIDFTP=FALSE"
    "-DCRYPTOPP_LOCATION=${cryptopp.dev}/include/cryptopp"
    "-DXROOTD_LOCATION=${xrootd.dev}/include/xrootd"
    "-DGTEST_LOCATION=${gtest.dev}/include/gtest"
  ];


  meta = with lib; {
    description = "Multi-protocol data management library";
    homepage = "https://cern-fts/gfal2";
    maintainers = [ ];
    license = licenses.asl20;
  };
}
```

**Comment by ShamrockLee** [comment]
Thank you for working on Nix packaging for gfal2!

I recently made `gfal2`, `gfal2-python` and `gfal2-util` into Nixpkgs (NixOS/nixpkgs#200100). It comes with per-plugin switch and functionality tests for XRootD-based source copying / directory reading (`gfal2-util.tests`). Feel free to test it, and add yourself as a package maintainer if you are willing to.

> * I have issues with gfal2 + xrootd + proxy certificate, which gives the following error:
>       ```
>       gfal-ls error: 52 (Invalid exchange) - [gfal2_stat][gfal_plugin_statG][gfal_xrootd_statG] Failed to stat file (Invalid exchange)
>       ```

Not sure if this is related to NixOS/nixpkgs#169677, which has been fixed by veprbl in NixOS/nixpkgs#289140. Feel free to try `gfal2`-related packages from the `nixos-unstable` branch and see if the issue also occurs there. If so, please kindly file an issue at [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs) and tag me.

**Comment by v1kko** [comment]
Thank you, the packages are working for me, and the error i had is now fixed!
